src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useContext, useCallback } from 'react';
+import {
+  TodoContext, TodoUpdateContext,
+} from '../TodoContext';
+import { Navigation } from './Navigation';
+
+export const Footer: React.FC = React.memo(() => {
+  const { todos } = useContext(TodoContext);
+  const { deleteTodos, setActiveIds } = useContext(TodoUpdateContext);
+  const isCompleted = todos.some(todo => todo.completed);
+  const items = todos.filter(todo => !todo.completed).length;
```

Comment:
don't forget about using useMemo and useCallback

Reply:
I thought I wrap them, will fix it now